File: Shared/Requests/RadixRequest.cs
namespace Shared.Requests
{
public class RadixRequest : BaseRequest
{
    public List<string> Numbers { get; set; } = [];

    public RadixRequest()
    {
    }

    public RadixRequest(RadixRequest request) : base(request)
    {
        Numbers = request.Numbers;
    }
}
}

File: Web/src/models/requests/RadixRequest.ts
import { BaseRequest } from "./BaseRequest";

export class RadixRequest extends BaseRequest {
  public numbers: string[] = [];

  public constructor(from: string, to: string, numbers?: string[]) {
    super(from, to);

    if (numbers) {
      this.numbers = numbers;
    }
  }

  public override Initialize(data: string[]) {
    if (!data) {
      throw new Error("The numbers are null!");
    }

    this.numbers = data;
  }
}
